### PR TITLE
add the ability to configure inputs.exec error truncation

### PR DESCRIPTION
**Pull request by ssoroka**

### CODE REVIEW

File: plugins/inputs/exec/README.md [hunk 1]
```diff
@@ -22,6 +22,9 @@ This plugin can be used to poll for custom metrics from any source.
   ## measurement name suffix (for separating different commands)
   name_suffix = "_mycollector"
 
+  ## truncate errors after this many characters. Use 0 to disable error truncation.
+  # error_truncate_length = 512
```
Comment (srebhan) [inline plugins/inputs/exec/README.md:26]: Shouldn't this be zero to keep the previous behavior?

File: plugins/inputs/exec/exec.go [hunk 2]
```diff
@@ -30,7 +30,10 @@ const sampleConfig = `
   timeout = "5s"
 
   ## measurement name suffix (for separating different commands)
-  name_suffix = "_mycollector"
+	name_suffix = "_mycollector"
```
Comment (srebhan) [inline plugins/inputs/exec/exec.go:33]: There seems to be a formatting problem...

File: plugins/inputs/exec/exec.go [hunk 3]
```diff
@@ -39,12 +42,13 @@ const sampleConfig = `
   data_format = "influx"
 `
 
-const MaxStderrBytes = 512
+const MaxStderrBytes int = 512
```
Comment (srebhan) [inline plugins/inputs/exec/exec.go:45]: What happens if the user specifies more than this max? You maybe should check that in `Init()`, clip to max and issue a warning...

**Review by srebhan — COMMENTED** [review]
I only have a minor comment (see code). Could you please also add a test-case where you actually truncate with a non-default argument!? Just to make sure... :-)

**Comment by srebhan** [comment]
@ssoroka ping... ;-P

**Comment by srebhan** [comment]
@ssoroka any news on this?

**Comment by helenosheaa** [comment]
When Dave and I paired on reviews we were wondering what problem this is trying to solve? When would the user know that they need this?

File: plugins/inputs/exec/exec.go (hunk 2 above)
Comment (ssoroka) [reply plugins/inputs/exec/exec.go:33]:
```suggestion
  name_suffix = "_mycollector"
```

**Review by sspaink — APPROVED** [review]
lgtm

**Comment by ssoroka** [comment]
exec truncates the output by default, so you might not be able to see the whole message about what is causing the script to fail. There's an open issue or user request for this somewhere; not sure why it's not linked.

An alternative (or corollary) to this is to show the whole output when running in debug, and then adjusting the length is less of an issue.

**Pull request merged**